src/db/roomstore.ts

```diff
@@ -0,0 +1,347 @@
+        WHERE ${whereClaues}`;
+        // tslint:disable-next-line no-any
+        return (await this.db.All(sql, data as any)).map((row) => {
+            const id = row.id as string;
+            const matrixId = row.matrix_id;
+            const remoteId = row.room_id;
+            return {
+                id,
+                matrix: matrixId ? new MatrixStoreRoom(matrixId as string) : null,
+                // tslint:disable-next-line no-any
+                remote: matrixId ? new RemoteStoreRoom(remoteId as string, row as any) : null,
+            };
+        });
+    }
+
+    public async removeEntriesByRemoteRoomId(remoteId: string) {
+        await this.db.Run(`DELETE FROM room_entries WHERE remote_id = $remoteId`, {remoteId});
+        await this.db.Run(`DELETE FROM remote_room_data WHERE room_id = $remoteId`, {remoteId});
+    }
+
+    public async removeEntriesByMatrixRoomId(matrixId: string) {
+        await this.db.Run(`DELETE FROM room_entries WHERE matrix_id = $matrixId`, {matrixId});
+        await this.db.Run(`DELETE FROM remote_room_data WHERE room_id = $matrixId`, {matrixId});
+    }
+
+    private async upsertRoom(room: RemoteStoreRoom) {
+        const existingRow = await this.db.Get(
+            "SELECT * FROM remote_room_data WHERE room_id = $id",
+            {id: room.roomId},
+        );
+
+        if (!room.data) {
+            throw new Error("Tried to upsert a room with undefined data");
+        }
+
+        const data = {
+            discord_channel:     room.data.discord_channel,
+            discord_guild:       room.data.discord_guild,
+            discord_iconurl:     room.data.discord_iconurl,
+            discord_iconurl_mxc: room.data.discord_iconurl_mxc,
+            discord_name:        room.data.discord_name,
+            discord_topic:       room.data.discord_topic,
+            discord_type:        room.data.discord_type,
+            plumbed:             room.data.plumbed || 0,
+            update_icon:         room.data.update_icon || 0,
+            update_name:         room.data.update_name || 0,
+            update_topic:        room.data.update_topic || 0,
+        } as IRemoteRoomData;
+
+        if (!existingRow) {
+            // Insert new data.
+            await this.db.Run(
+            `INSERT INTO remote_room_data VALUES (
+                $id,
+                $discord_guild,
+                $discord_channel,
+                $discord_name,
+                $discord_topic,
+                $discord_type,
+                $discord_iconurl,
+                $discord_iconurl_mxc,
+                $update_name,
+                $update_topic,
+                $update_icon,
+                $plumbed
+            )
+            `,
+            {
+                id: room.roomId,
+                // tslint:disable-next-line no-any
+                ...data as any,
+            });
+            return;
+        }
+
+        const keysToUpdate = { };
```

Comment:
this doesn't have a type

Reply:
I suspect I was being lazy, this should probably be `IWhatItsNameLazy`